optuna/multi_objective/trial.py

```diff
@@ -0,0 +1,218 @@
+
+    def suggest_uniform(self, name: str, low: float, high: float) -> float:
+        return self._trial.suggest_uniform(name, low, high)
+
+    def suggest_loguniform(self, name: str, low: float, high: float) -> float:
+        return self._trial.suggest_loguniform(name, low, high)
+
+    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
+        return self._trial.suggest_discrete_uniform(name, low, high, q)
+
+    def suggest_int(self, name: str, low: int, high: int) -> int:
+        return self._trial.suggest_int(name, low, high)
+
+    def suggest_categorical(
+        self, name: str, choices: Sequence[CategoricalChoiceType]
+    ) -> CategoricalChoiceType:
+        return self._trial.suggest_categorical(name, choices)
+
+    def report(self, values: List[float], step: int) -> None:
+        if len(values) != self._n_objectives:
+            raise ValueError(
+                "The number of the intermediate values {} at step {} is mismatched with"
+                "the number of the objectives {}.",
+                len(values),
+                step,
+                self._n_objectives,
+            )
+
+        for i, value in enumerate(values):
+            self._trial.report(value, self._n_objectives * (step + 1) + i)
+
+    def _report_complete_values(self, values: List[float]) -> None:
+        if len(values) != self._n_objectives:
+            raise ValueError(
+                "The number of the values {} is mismatched with the number of the objectives {}.",
+                len(values),
+                self._n_objectives,
+            )
+
+        for i, value in enumerate(values):
+            self._trial.report(value, i)
+
+    def set_user_attr(self, key: str, value: Any) -> None:
+        self._trial.set_user_attr(key, value)
+
+    def set_system_attr(self, key: str, value: Any) -> None:
+        self._trial.set_system_attr(key, value)
+
+    @property
+    def params(self) -> Dict[str, Any]:
+        return self._trial.params
+
+    @property
+    def distributions(self) -> Dict[str, BaseDistribution]:
+        return self._trial.distributions
+
+    @property
+    def user_attrs(self) -> Dict[str, Any]:
+        return self._trial.user_attrs
+
+    @property
+    def system_attrs(self) -> Dict[str, Any]:
+        return self._trial.system_attrs
+
+    @property
+    def datetime_start(self) -> Optional[datetime]:
+        return self._trial.datetime_start
+
+    # TODO(ohta): Add `to_single_objective` method.
+    # This method would be helpful to use the existing pruning
+    # integrations for multi-objective optimization.
+
+    @property
+    def _values(self) -> List[Optional[float]]:
+        trial = self._trial.study._storage.get_trial(self._trial._trial_id)
+        return [trial.intermediate_values.get(i) for i in range(self._n_objectives)]
+
+
+@experimental("1.4.0")
+class FrozenMultiObjectiveTrial(object):
```

Comment:
This is not a strong opinion, but I like the current naming rule because I can interpret, for example, `BaseMultiObjectiveSampler` as "the **base** interface of **multi-objective** **sampler**s." It's a bit more natural to me than `MultiObjectiveBaseSampler` that seems like "the **multi-objective** version of the **base** **sampler**."
(especially if we implement multi-objective samplers that have no single-objective alternative like NSGA-II in the future)

However, I would like to hear opinions from others. If most of the people prefer the naming rule like `BaseMultiObjectiveSampler`, I'm willing to adopt it.

Reply:
I see, you are right about the base class naming. The current naming makes sense considering that. I also tried to interpret the semantic hierarchy, i.e. is it "a frozen trial that is a multi-objective" (this comments) or is it "a multi-objective trial that's frozen" (current) and your convention seem more intuitive.